pkg/trace/obfuscate/sql.go

```diff
@@ -248,15 +283,18 @@ type ObfuscatedQuery struct {
 // attemptObfuscation attempts to obfuscate the SQL query loaded into the tokenizer, using the
 // given set of filters.
 func attemptObfuscation(tokenizer *SQLTokenizer) (*ObfuscatedQuery, error) {
+	storeTableNames := config.HasFeature("table_names")
+	normalizeTables := config.HasFeature("normalize_sql_tables")
+
 	filters := []tokenFilter{
 		&discardFilter{},
-		&replaceFilter{},
-		&groupingFilter{},
 	}
-	tableFinder := &tableFinderFilter{}
-	if config.HasFeature("table_names") {
+	tableFinder := &tableFinderFilter{storeTableNames: storeTableNames}
+	if storeTableNames || normalizeTables {
 		filters = append(filters, tableFinder)
```

Comment:
If we're not storing table names, why are we adding this for normalizing traces? Also, no need to initialise the variable outside the `if` statement right?

Reply:
The normalization only happens on tables, which requires the identification of the `Table` identifier. To avoid doing this twice when both features are turned on, the identification of `Table` happens solely in the table finder filter.

The variables are used twice: once to pass into the filter structs, again to the if condition.